lib/widgets/recent_dm_conversations.dart

```diff
@@ -38,21 +53,40 @@ class _RecentDmConversationsPageState extends State<RecentDmConversationsPage> w
     });
   }
 
+  void _unreadsModelChanged() {
+    setState(() {
+      // The actual state lives in [unreadsModel].
+      // This method was called because that just changed.
+    });
+  }
+
   @override
   Widget build(BuildContext context) {
     final sorted = model!.sorted;
+    final unreadDms = unreadsModel!.dms;
     return Scaffold(
       appBar: AppBar(title: const Text('Direct messages')),
       body: ListView.builder(
         itemCount: sorted.length,
-        itemBuilder: (context, index) => RecentDmConversationsItem(narrow: sorted[index])));
+        itemBuilder: (context, index) {
+          final narrow = sorted[index];
+          return RecentDmConversationsItem(
+            narrow: narrow,
+            unreadCount: unreadDms[narrow]?.length ?? 0,
```

Comment:
:eyes: on this, I've got https://github.com/sirpengi/zulip-flutter/commit/237ea8fee8fbf3bf29a4f663a2c52e7a4b942c38 for my own usage in my work on #130

Reply:
> It occurs to me that it might be ideal for the pieces of the unreads model like `dms` and `streams` to be private, or `@visibleForTesting`. But we can see whether that feels right as we add more things consuming this model.

I think we'll want the unreads model to expose unread IDs so that UI code can do some filtering based on mute state.

Reply:
> I think we'll want the unreads model to expose unread IDs so that UI code can do some filtering based on mute state, which will be tracked elsewhere.

Hmm yeah, good point.

Reply:
For the counts that require traversing the model and doing sums (so, total unreads and unreads per stream), I wonder about instead adding properties that the model maintains, and updating when the model handles events etc. Then UI-code callers don't have to worry about a lot of repetitive counting work being done.

I've sent the draft #345 for one approach to tracking those numbers.